BrentOzarULTD/SQL-Server-First-Responder-Kit
# #293 Support for @OutputServerName in sp_BlitzCache

Fixes #293 for sp_BlitzCache.

Changes proposed in this pull request:
- Support for @OutputServerName
- Updated error checking for @OutputDatabaseName variables for both linked and local servers

How to test this code:
- Run procedure output with @expertmode = 1 (for comparison)
- Run procedure with valid @OutputDatabaseName/SchemaName/TableName specified (should succeed in creating local table)
- Run procedure with @OutputServerName specified as valid linked server along with other Output variables (should succeed in creating remote table on linked server)
- Rerun both tests above without deleting testing tables to ensure that procedure doesn't try and create new table (it should dump results to the same table; tests IF EXISTS logic in permanent table creation)
- Run procedure with invalid @OutputServerName (server that is not linked and random name; both should fail)
- Run procedure with invalid local DB name (should fail)
- Run procedure with invalid linked DB name (Specify valid @OutputServerName; should fail)

Use the output from the @expertmode = 1 run to compare the output on both the local and remote tables...they should be the same.

Has been tested on:
- SQL Server 2008
- SQL Server 2008 R2
- SQL Server 2012
- **(UPDATE)** SQL Server 2016
